tests/service_controller_test.go

```diff
@@ -0,0 +1,75 @@
+package tests
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"kubevirt.io/ssp-operator/controllers"
+	"kubevirt.io/ssp-operator/internal/operands/metrics"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getSspMetricsService() (*v1.Service, error) {
+	service := controllers.ServiceObject(strategy.GetSSPDeploymentNameSpace())
+	err := apiClient.Get(ctx, client.ObjectKeyFromObject(service), service)
+	return service, err
+}
+
+func equalService(serviceA, serviceB *v1.Service) bool {
+	return reflect.DeepEqual(serviceA.Labels, serviceB.Labels) && reflect.DeepEqual(serviceA.Spec, serviceB.Spec)
+}
+
+var _ = Describe("Service Controller", func() {
+	var (
+		serviceErr error
+		service    *v1.Service
+	)
+	BeforeEach(func() {
+		waitUntilDeployed()
+	})
+
+	It("[test_id: 8807] Should create ssp-operator-metrics service", func() {
+		service, serviceErr = getSspMetricsService()
+		if serviceErr != nil {
+			Fail("Failed to get ssp-operator-metrics service")
+			return
+		}
+	})
+
+	It("[test_id: 8808] Should re-create ssp-operator-metrics service if deleted", func() {
+		service, serviceErr = getSspMetricsService()
```

Comment:
Same as above, please use `Expect` to check the error.

Reply:
yes, forgot to add it after moving this line of code out of before each to three places :)